File: 938. Range Sum of BST/main.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def rangeSumBST(self, root: TreeNode, L: int, R: int) -> int:
        # method 1:
        if not root:
            return 0
        if (root.val < L):
            return self.rangeSumBST(root.right,L,R)
        if (root.val > R):
            return self.rangeSumBST(root.left,L,R)
        return root.val+self.rangeSumBST(root.left,L,R)+self.rangeSumBST(root.right,L,R)
        
        # method 2:
        # if not root:
        #     return 0
        # else:
        #     value = 0
        #     if L<=root.val<=R:
        #         value = root.val
        #     else:
        #         pass
        #     return value +self.rangeSumBST(root.left,L,R) + self.rangeSumBST(root.right,L,R)

File: 938. Range Sum of BST/main.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int rangeSumBST(TreeNode root, int L, int R) {
        // method 1
        if (root==null) return 0;
        if(root.val <L){
            return rangeSumBST(root.right,L,R);
        }
        if (root.val >R){
            return rangeSumBST(root.left,L,R);
        }
        return root.val+rangeSumBST(root.left,L,R)+rangeSumBST(root.right,L,R);
        
        // method 2
        // if (root==null) return 0;
        // else {
        //     return (L<=root.val && root.val<=R ? root.val:0) + rangeSumBST(root.left,L,R) + rangeSumBST(root.right,L,R);
        // }
    }
}